### ISSUE TITLE
Firefox's Password Manager should work with HTTP authentication dialog

### ISSUE THREAD

**Issue by yookoala** [mozilla-mobile/fenix#8642]
### Why/User Benefit/User Problem
It works on Firefox desktop and the Fennec. And it is kind of expected by users.

### What/Requirements
Password Manager should work to remember HTTP authentication username / password just as it does for normal HTML login forms.

### Acceptance Criteria (how do I know when I’m done?)
1. If a password of the HTTP authentication site is in the Password Manager, Fenix should display the suggested username and password on the popup login dialog.
2. If a password of the HTTP authentication site is NOT in the Password Manager, Fenix should wait until user entry then ask if she / he wants to store the login information to the Password Manager.

**Comment by yookoala** [comment]
A duplication of #4992. Close now.

**Comment by ekager** [comment]
I'm reopening this because it is about the internal password manager working with HTTP Authentication Dialogs and different than #4992

**Comment by TroelsS** [comment]
More info on this bug here: https://bugzilla.mozilla.org/show_bug.cgi?id=993902

**Comment by TroelsS** [comment]
I do not think it is a "feature request" but rather a bug.

This worked fine in Firefox 68.0.1.

**Comment by fichtennadel** [comment]
see also https://github.com/mozilla-mobile/fenix/issues/9257

**Comment by yookoala** [comment]
Still doesn't work when it rolled out 😟

**Comment by mpopp75** [comment]
This clearly is a bug and should be given higher priority.

**Comment by Antagony1060** [comment]
Why is this still flagged as a ‘feature request’? It is clearly a bug as the behaviour has regressed between versions. 

This is a real pain for me as I use HTTP authentication extensively for testing and I don't want the hassle of having to fire up my separate password manager every time I want to check them via a mobile device.

**Comment by Dunexus** [comment]
Fenix is a entire rewrite of the browser around GeckoView.
As this feature has not been implemented in Fenix, it is a feature request (code that need to be written) and not a regression (code that worked and doesn't anymore).
The labels are from a developer's point of view to help them knowing where to look. This labeling choice is **not** used to deny the fact that this feature was present in Fennec (old-firefox) and has not been ported yet.

**Comment by WeirdConstructor** [comment]
From a user perspective this is a regression, as I just used to use Firefox on my mobile and from one day to the next it stopped working. Wish I could go back to the older Firefox version easily. As developer I understand that this is a feature "request" in case of a rewrite. As user I don't understand why a rewrite was necessary in the first place. And it breaks something I use on a daily basis. So I might have to go back to chrome for now.

**Comment by Antagony1060** [comment]
>From a user perspective this is a regression, as I just used to use Firefox on my mobile and from one day to the next it stopped working.

Quite so @WeirdConstructor. The obvious concern for those of us objecting to the labelling/handling of this issue is that as a ‘feature request’ it will no doubt have a lower priority and therefore either take a long time to get fixed or may be ignored altogether. The very fact it still has no assignee fully seven months after it was opened would seem to vindicate that fear.

**Comment by aslmx** [comment]
This is still the case in recent Firefox builds. 
It stopped working a few months ago, all of a sudden.

I wish this could be given the appropriate BUG label (or regression, as it has been pointed out often enough that this has worked just fine before) and priority.

It is utterly annoying.

**Comment by yookoala** [comment]
> This is still the case in recent Firefox builds.
> It stopped working a few months ago, all of a sudden.
> 
> I wish this could be given the appropriate BUG label (or regression, as it has been pointed out often enough that this has worked just fine before) and priority.
> 
> It is utterly annoying.

On July this year, the official release of "Firefox for Android" has switched from the old [Fennec](https://wiki.mozilla.org/Mobile/Fennec) to a completely rewritten one, codename [Fenix](https://github.com/mozilla-mobile/fenix). Since this is a complete rewrite, the development team considers this previously Fennec-supported feature as a "new feature" in Fenix.

This is a bit ridiculous in users' perspective, but this is why this issue is not labelled a "bug".

**Comment by gusowski1** [comment]
But from an user perspective it is a bug and a reason to switch the browser
to get things working as expected.
I switched my primary mobile browser because of this reason weeks ago and
up to now i didn't switched back cause of several things not working
anymore.

I understand the behavior from a developer perspective - but from the
userpoint it is a nogo.
But it is a good way to lose more and more users.
But probably mozilla's user base has grown in sum within the last month
more as the lose of old fennec users cause of the switch from fennec to
fenix.



Am Mi., 7. Okt. 2020 um 14:50 Uhr schrieb Koala Yeung <
notifications@github.com>:

**Comment by yookoala** [comment]
@gusowski1: No objection here. I think this is a bug / regression for me.

**Comment by Hubfront** [comment]
It's a bug, if the internal password manager can't handle basic authentification.

**Comment by riksoft** [comment]
I confirm this huge bug is still in the any firefox for Android available to date, even on the nightly.
This bug forces the user to retype user/pass every time.

**Apparently there is also another bug**
that prevents even the workaround of passing user/password via url in the form user:pass@host that is a standard format working on firefox desktop or any other browser mobile.

**Comment by Hubfront** [comment]
@riksoft Thats right. Firefox devs need to know, if they bother at all: it's not possible to copy the login credentials from the login credential page of fenix, because the login prompt will be gone already if you got back to the basic auth page. Simply reloading wont help either, because fenix wont bring the prompt back again, until you close the browser entirely with all opened tabs and sessions, and till you open the basic auth page again to insert the login credentials. But then you have only a copy of one of the two credentials: user or pass. If you choose to copy the password, what most of us will probably do, because the pass should be not easy to remember, you need to remember the username from the brain in order to fill it in the login prompt, paste the passwort and it was so easy and user friednly of the shiny new fenix, wasnt it .. to get us logged in finally. :D

Maybe developement thinks, that we expect to much and that we should be thankful, that the internal passwort manager of fenix is able to fill in the appropriate login credentials when other authentification methods are used, at least after one year of "feature request" .. since August 2020.

**Comment by Hubfront** [comment]
No one assigned after 9 months. Same is the case for the bug that addresses autofill of http auth dialog by 3rd party password manager.

Thats how you build non-competitive software.

**Comment by WeirdConstructor** [comment]
I am kind of relieved that I am not the only one relying on basic auth. And I also tried http://user:pass@domain/... - Chrome handles all this a little bit better.

I don't believe there is any malicious intend with the priority of this bug or missing feature.
Developers are usually trying their best, but are all bound to the resources and time that are allocated for them to work on issues/tasks. For some that is time they are paid to work on it, and for some it's time they are willing to invest without any compensation.

**Comment by ekager** [comment]
@Hubfront this is a warning that your behavior here is not acceptable. Please familiarize yourself with [the CPG](https://www.mozilla.org/en-US/about/governance/policies/participation/) if you would like to interact with the team in this repo.

I have filed a Bugzilla ticket to discuss options here if you would like to follow along. 
https://bugzilla.mozilla.org/show_bug.cgi?id=1676216 

@riksoft Please open other bugs in new tickets, I'm not sure I understand your comment above about pasting

**Comment by Hubfront** [comment]
@it's not my intend to interact with the devs in the first place, but to improve fenix to make it a better browser.

**Comment by Hubfront** [comment]
Its not my intent to go on somebodys nerves. I want fenix to get a better browser. It goes on my nerves, if i cant log in.

**Comment by ekager** [comment]
And I hear you that it gets on your nerves, but it is never helpful or acceptable to use language like "lazy developers" and "Thats how you build non-competitive software". 

We have a very small team and almost 3k open issues. I'm sure everyone thinks the issue that bothers them the most is the most important.

**Comment by Hubfront** [comment]
I use firefox since some time and trust the team and their integrity more than the ones of other browser companies.

**Comment by Hubfront** [comment]
@ekager ill follow that the opened issued on bugzilla with interest. It seems that there was another open bug on bugzilla (7 years old, unfixed), that adresses the same problem with fennec 31 and was moved to github three months ago. It points to #8642

**Comment by riksoft** [comment]
@ekager "I'm not sure I understand your comment above about pasting"
I said "passing". To work around this problem I tried an intermediate page as a redirect to the target auth page, and this intermediate page was passing (get instead of post) a url in the form of **user:pass@host**. Unfortunately that doesn't work either (despite the desktop version of Firefox, and any other mobile browser work ok both with basich auth and this kind of URL).

I can't wait till this bug will be eventually solved, so I've though another way to work around this problem and I've found one perfect for my case. Hope this could be of use to others: I've "whitelisted" the LAN IPs in the same .htaccess used for the basic auth, so I can skip the authentication altogether when I'm accessing the page from the LAN or via SSH from remote.

`AuthType Basic`
`AuthName "Secure Area"`
`AuthUserFile /xxxxx/xxxx/.htpasswd`
`Require valid-user`
`Require ip 192.168.0`

That's not OK for any situations (e.g. not for a website accessed directly via http) but for local servers or http via SSH can do the job.

**Comment by aslmx** [comment]
> I've "whitelisted" the LAN IPs in the same .htaccess used for the basic auth, so I can skip the authentication altogether when I'm accessing the page from the LAN or via SSH from remote.

I am doing exactly this for many years now, so that i don't have to fiddle around when i'm at home (or at work, where i SSH tunnel into my home network anyway).

However, this does not help in all cases. E.g. sometimes the WiFi seems active, but still Firefox (Android) asks me for basic auth. The connection seems to be stuck on mobile data then. This might be a problem of my smartphone, but anyways... **This is just a workaround no solution.**

I'm almost thinking about using client side certs to protect my selfhosted services. But i wouldn't really expect this to be working flawlessly when basic auth credentials can neither be stored nor be passed via URI...

**Comment by Hubfront** [comment]
@riksoft tx for sharing your workaround with us. You mentioned also, that any other mobile browser can both handle basic auth and passing the credentials via url like user=xxx:pass=xxx@host

In order to keep firefox for Android competitive, fenix should be able to handle these very basic autgentification methods too. Because, for the average user, say an studebt trying to access an restricted document on the server of his university, its frustrating to have all these credentials in the firefox database but they are useless, because the internal passwort manager cant use them when basic auth is used.

**Comment by riksoft** [comment]
@aslmx @Hubfront 
No doubt that mine is just a desperate workaround for personal stuff: I can't wait months or years for the real solution. However 
Firefox mobile should surely manage standard protocols like basic auth or the url user:pass@host.

**Comment by Hubfront** [comment]
I hope well see progress in January 2021.

**Comment by aslmx** [comment]
Almost another 8 weeks passed. 
I'm away from my LAN over christmas and everytime i access webpages hosted at home, I hit the .htaccess basic auth. I'm seriously annoyed.

If we can't get a date for the fix, can we get a date when this will be decided?

**Comment by Hubfront** [comment]
> 
> 
> Almost another 8 weeks passed.
> I'm away from my LAN over christmas and everytime i access webpages hosted at home, I hit the .htaccess basic auth. I'm seriously annoyed.
> 
> If we can't get a date for the fix, can we get a date when this will be decided?

Merry christmas to you. I would expect it to be fixed in Q1, thats what Eugen Sawin said. He works since 7 years for Mozilla and seems to have some influence in the developent team.. If we would see some progress in January on this would be gladly appreciated.

**Comment by Hubfront** [comment]
Any update on basic authentification?

**Comment by Hubfront** [comment]
When will someone fix this bug?

**Comment by aslmx** [comment]
I'll prepare a shopping list for a birthday cake... this issue is turning 1 in a few days...

**Comment by Hubfront** [comment]
aslmx this will be a sad birthday. 😞

**Comment by FFranta** [comment]
Wow finally I was able to find I am not alone with this problem. I think I have the problem since about August 2020. Do you have any information about when it could be fixed? Thanks.

**Comment by ekager** [comment]
Sorry for the frustration. Please follow along in the Bugzilla bug since this cannot be fixed in Fenix alone https://bugzilla.mozilla.org/show_bug.cgi?id=1676216

**Comment by yookoala** [comment]
> Sorry for the frustration. Please follow along in the Bugzilla bug since this cannot be fixed in Fenix alone https://bugzilla.mozilla.org/show_bug.cgi?id=1676216

It's good to know that the issue is on some kind of the Q1 roadmap. Thanks [Eugen](https://bugzilla.mozilla.org/user_profile?user_id=495422) for working on it and @ekager for updating. As a long time Firefox user, all I hope for was some updates. I believe you're doing things as fast as can be.

**Comment by Hubfront** [comment]
> Sorry for the frustration. Please follow along in the Bugzilla bug since this cannot be fixed in Fenix alone https://bugzilla.mozilla.org/show_bug.cgi?id=1676216

There isnt an update since thre months. This bug is open more than a year.

**Comment by eliserichards** [comment]
Looks like this is on the GV docket for Q1! We might consider closing this bug in favor of a fresh one once that work is done.

**Comment by Hubfront** [comment]
I would prefer to leave this bug open. If its closed, then it would be harder for us end-users to express our concern, that this bug is still not fixed in reality.

**Comment by Hubfront** [comment]
How will firefox be competitive with chrome, if the devs cant even fix basic authentification?

**Comment by majurgens** [comment]
I also want this "feature".  Depending on how problematic it is for you you could revert to using Firefox mobile 68, which is what I am now considering doing

**Comment by yookoala** [comment]
> Looks like [this](https://bugzilla.mozilla.org/show_bug.cgi?id=1676216) is on the GV docket for Q1! We might consider closing this bug in favor of a fresh one once that work is done.

@eliserichards: Sorry to be a bit pushy, but is there any news on any progress? As you mentioned before, this should be on GeckoView docket for Q1. It's close to the mid-Q2, has any developer picked this up yet? 🙏🏻

**Comment by Hubfront** [comment]
When will the reluctant developers fix this bug?

**Comment by susuper** [comment]
It was a super-feature of mobile FF. Looks like the developers have lost a link with the reality a bit(

**Comment by Hubfront** [comment]
Why i use firefox? Its not about features, wether its desktop or wether its about android. I trust the integrity of the organisation. So all my passwords are stored in firefox. It was therefore important to finally make the passwords usable in Android firefox in standard scenarios, but its also important to make the passwords usable in basic authentification. Because its about trust not about features, where mozilla shines.

**Comment by FFranta** [comment]
That is why I used Firefoxed until yesterday as well. Yesterday Firefox came with new android app design instead of fixing this bug which is there for more than a year.
Since yesterday I have been using Chrome and everything is ok - all passwords, bookmarks etc were imported correctly and PC/mobile sync works, as well as passwords stored for basic http authentication. I was paitiently waiting 16 months, hoping someone will fix it.

**Comment by Hubfront** [comment]
Its not good, that chrome can handle multiple logins with basic auth while android firefox still cant - it cant even handle stored logins at all with this auth method. The use of multiple logins is a must have for mobile browsers since several years and firefox still lacks support for it with basic auth.

**Comment by dia-de-muertos** [comment]
Any news on this?
Is work in progress?
I'm still on firefox 68.11, but it's getting more and more dangerous, as it's doesn't get security fixes :-(

**Comment by ocroquette** [comment]
> I also want this "feature". Depending on how problematic it is for you you could revert to using Firefox mobile 68, which is what I am now considering doing

Do you know a reliable source to get Firefox mobile 68 and this feature back ?

**Comment by majurgens** [comment]
> Do you know a reliable source to get Firefox mobile 68 and this feature back ?
I used one of the APKs here: https://www.edcint.co.nz/tmp/expire_14/Firefox68Android/
I can't recall which one. I'm not sure exactly what devices or Android versions they will work on but they do work on Android 8 Samsung Galaxy S7.
Most of my devices I just use the new version of Firefox but I regularly use sites with basic auth - and so I reverted to v68 on that device only.

The files will be gone in 14 days

I find many software developers prioritise new flashy features and redesigns over other functional features they already had. I don't quite understand why but they do. I guess they think that flashy new features and redesigns will attract more users than they will lose by dropping functionality.

**Comment by denebolar** [comment]
Wow, thanks, I was desperately looking for the good old fennec. I am sorry, guys, I support @majurgens comment on flashyness over functionality. I, myself, use http basic auth quite a lot in my setups. The lack of using the credentials manager there keeps me from using fenix. I deem this a core functionality which is not implemented, yet.

**Comment by ocroquette** [comment]
Thanks guys, based on this input I found the file also on Mozilla's servers:

https://ftp.mozilla.org/pub/mobile/releases/68.11.0/android-api-16/multi/fennec-68.11.0.multi.android-arm.apk

```
MD5 916984779fe804417a261e9a675a2b8b
SHA1 9225c1110b6102042eae838f92de697c13c8e4c3
```

**Comment by Hubfront** [comment]
Inteteresting. Will it work in parallel to an existing fenix installation?

**Comment by ocroquette** [comment]
No, without hacking around, one version can be installed at a time. One way to install the APK is to enable the development mode of the phone and push it using "adb" from a computer.

**Comment by peterv1988** [comment]
@eliserichards is there any update on this and if it could be relabeled to `bug` instead as mentioned in other replies? Maybe I'm not enough familiar with the issue system for Firefox on Github but it's not exactly clear to me what external blocking element here the assigned `waiting` label is referring to

**Comment by Hubfront** [comment]
I dont think to label this bug is enough. Devs are paid to fix a bug.

**Comment by kevincox** [comment]
Can we lock this to collaborators? These unhelpful comments are creating a lot of noise for people who just want to know when it is finished or progressing. Unfortunately this may make it difficult for people who want to contribute but locking seems like the more attractive option in this case.

**Comment by Hubfront** [comment]
This bug was opened in Feb 2020 and still no progress. I think, that critic is appropriate in this case.

**Comment by Hubfront** [comment]
This bug was opened in Feb 2020 and still no progress. Critic from user standpoint is appropriate and advisable in this case.

**Comment by Hubfront** [comment]
This bug was opened in Feb 2020 and still no progress. Critic from user perspective is therefore appropriate and advisable in this case. Chrome mobile can handle password management with basic auth, and in many environments, basic auth is used (e.g. many university sites are using it). Its not possible for a user to type in complicated passwords manually every time to log in to a basic auth protected website, because firefox password manager cant handle it (bug!). I dont want to switch to chrome mobile and i dont want this bug to be still open in 2022.

**Comment by yookoala** [comment]
@Hubfront: As the original poster of this issue and a fellow Firefox user, I understand all the frustrations with it. But I have to agree with @kevincox. Many of us (I'd say mostly NOT Mozilla developers) who subscribes to this want to know the updates about it. And since Mozilla has but a small team of developers on Fenix, I don't thing they'd put attention tracing every comments in every issues (including this one).

We all share the needs to this feature and the emotions about how this is handled. And we all want this to move forward. Having another angry comment is not exactly helpful to resolving the issue. And I believe enough people have expressed how this is affecting them. Another repeated comment would just be another notification to all (and probably not noticed by the developers).

If you want to see how things are going, please keep track of [https://bugzilla.mozilla.org/show_bug.cgi?id=1676216](bug 1676216) on Bugzilla. As far as I can see, one developer is currently working on this issue. If you really want to express anything to him, please be polite.

**Comment by yookoala** [comment]
> @eliserichards is there any update on this and if it could be relabeled to `bug` instead as mentioned in other replies?
> 
> Maybe I'm not enough familiar with the issue system for Firefox on Github, but it's not exactly clear to me what external blocking element here the assigned `waiting` label is referring to? **Edit**: for reference, referring to:
> <img alt="afbeelding" width="506" src="https://user-images.githubusercontent.com/19436743/127130796-cc853059-246e-4f99-ab76-c51517feb33f.png">

This issue is blocked by how [GeckoView](https://github.com/mozilla/geckoview) is handling Android's autofill. That's why it is marked "blocked" here.

**Comment by Hubfront** [comment]
It was "Deprioritised" 19 days ago. https://bugzilla.mozilla.org/show_bug.cgi?id=1676216#c4

**Comment by yookoala** [comment]
> It was "Deprioritised" 19 days ago. https://bugzilla.mozilla.org/show_bug.cgi?id=1676216#c4
> 
> @yookoala As i said before, no progress, only some labeling from time to time. Its advisable in my opinion, to critize this from time to time. It's time to praise, when there is progress.

@Hubfront: I do not wish to prolong this conversation. So this is probably my last comment for it.

I've never asked you to "praise" anybody. Nor have I asked you to stop criticizing. But if you want to express something, you'd need to find the right channel. And I think this is the __wrong__ platform for it. As I said before, there are only a handful of developers in the team. They're probably __NOT__ reading every, or any, new comment in this issue. Why here?

Say, if you're angry or frustrated about Emily Toop's decision to deprioritize the GV bug, you could send her an email or an open letter. You (or anybody) can even found a petition campaign, which I'd definitely sign to endorse. If you do, you should definitely post a link of your open letter / petition campaign here, too. I'm sure many of the subscribers here would endorse it. You can put it on Redit or any public outlet. There are many platform more suitable than a GitHub issue, where only a small group of affected fellow Firefox users are subscribed to.

Another comment of "me too", "this affected me too", or "I'm frustrated this take so long" here won't be helpful to you, me, or anybody. I believe we have enough of these said here. We all share the needs and frustrations. Another repeated comment does not make any progress.

**Comment by Hubfront** [comment]
@yookoala You are totally wrong. I surely dont and wont express praise or critic here, because you or somebody else expect it from me. I am not frustrated about the "Depriotizing", i just mentioned it to underline my standpoint, that there is no progress, just some labeling from time to time. And this is the right place for me to do it. Feel free to add me on your ignore list, if you dont like my critic on this matter.

**Comment by agi** [comment]
After Bug 1710668 lands, there's a dependency in AC: https://github.com/mozilla-mobile/android-components/issues/10867 and then this should work out of the box for Fenix.

**Comment by yookoala** [comment]
> It works! Good job. You really made my day. Was waiting for this issue to be fixed for more than a year. Switched to and back from Edge a couple of times in agony. Off to the Play Store to update my Firefox rating.

@rozmansi: Which version are you using now? I'm on 92 and still no luck yet.

**Comment by rozmansi** [comment]
@yookoala 93.1.0

**Comment by yookoala** [comment]
Since 93.1.0 has implemented the feature, I guess we can finally close this issue for good.

**Comment by tosiara** [comment]
I remember this was an issue few years ago, so I decided to install Firefpx for Android again and give it a try.
Even though `93.1.0` is mentioned as fixed, the latest version from Play Store is `104.2.0` and it still does not work with basic http auth as it worked previously. I have synced passwords, I can see a password in the list, but a s soon as I press on the saved url - it opens and there is a standard basic http auth prompt without auto-filled creds, completely empty. Now way to fill in the saved creds.

Could someone who marked this issue as solved confirm, how did you check? How does it work for you?

**Comment by RES-1** [comment]
> Could someone who marked this issue as solved confirm, how did you check? How does it work for you?

It works as expected. With HTTP(S) basic authentication, the user data is pre-filled and only needs to be confirmed.

**Comment by the2masters** [comment]
tried it right now on desktop and android, both work fine! Android Version 104.2.0

**Comment by tosiara** [comment]
Could you try to create a test login for this web site and make a screenshot how it is autofiled in Firefox Android?

https://linuks.lt/firefox_android_bug/

I get this:

![Screenshot_20220912-143022_Firefox_edited](https://user-images.githubusercontent.com/3600973/189643957-466e058e-1173-4772-b906-6452c6bb99a5.jpg)
![Screenshot_20220912-143147_Firefox](https://user-images.githubusercontent.com/3600973/189643961-62331997-62b0-4983-8c25-6edd589905ad.jpg)

**Comment by RES-1** [comment]
> Could you try to create a test login for this web site and make a screenshot how it is autofiled in Firefox Android?

![1](https://user-images.githubusercontent.com/4004046/189656917-41a7a4c9-b5d3-4d5f-aa04-df082f15502d.png)
![2](https://user-images.githubusercontent.com/4004046/189656991-877b6498-8da2-45e8-82e7-005a5e378405.png)

**Comment by tosiara** [comment]
Thank you for the test!

So now the question, why it does not auto-fill on my device? Any custom setting must be enabled?
I use stock Samsung Android 12, stock Firefox 104.2.0 from Play Store.

**Comment by tosiara** [comment]
Ok, I may have found the answer for my own question.
For some reason, you cannot auto-fill manually created passwords. Only those that Firefox has saved for you. because there is a hidden attribute that gets saved into Sync Account - `httpRealm`. 

@RES-1 Did you create the password manually by going to Setting->Logins->Saved logins->Add login?
Or did you typed-in the passwords once, and then select `Save this login`?
